src/gui/mainwindow.cpp

```diff
@@ -15,27 +22,81 @@ MainWindow::MainWindow(QWidget* parent)
     ui->centralWidget->layout()->setContentsMargins(0, 0, 0, 0);
 
     auto modelRegistry = std::make_shared<QtNodes::DataModelRegistry>();
+    modelRegistry->registerModel<CanDeviceModel>();
+    modelRegistry->registerModel<CanRawSenderModel>();
+    modelRegistry->registerModel<CanRawViewModel>();
 
-    canDevice = std::make_shared<CanDevice>();
     graphScene = std::make_shared<QtNodes::FlowScene>(modelRegistry);
 
+    connect(graphScene.get(), &QtNodes::FlowScene::nodeCreated, this, &MainWindow::nodeCreatedCallback);
+    connect(graphScene.get(), &QtNodes::FlowScene::nodeDeleted, this, &MainWindow::nodeDeletedCallback);
+    connect(graphScene.get(), &QtNodes::FlowScene::nodeDoubleClicked, this, &MainWindow::nodeDoubleClickedCallback);
+
     setupMdiArea();
     connectToolbarSignals();
     connectMenuSignals();
+}
+
+MainWindow::~MainWindow() { delete graphView; }
+
+void MainWindow::closeEvent(QCloseEvent*) { handleExitAction(); }
+
+void MainWindow::nodeCreatedCallback(QtNodes::Node& node)
+{
+    auto dataModel = node.nodeDataModel();
 
-    canDevice->init("socketcan", "can0");
-    canDevice->start();
+    if (dataModel->name() == "CanRawSenderModel") {
+
+        auto rawSender = &static_cast<CanRawSenderModel*>(dataModel)->canRawSender;
+        ui->mdiArea->addSubWindow(rawSender);
+        connect(rawSender, &CanRawSender::dockUndock, this, [this, rawSender] { handleDock(rawSender, ui->mdiArea); });
+
+    } else if (dataModel->name() == "CanRawViewModel") {
+        auto rawView = &static_cast<CanRawViewModel*>(dataModel)->canRawView;
+        ui->mdiArea->addSubWindow(rawView);
+        connect(ui->actionstart, &QAction::triggered, rawView, &CanRawView::startSimulation);
+        connect(ui->actionstop, &QAction::triggered, rawView, &CanRawView::stopSimulation);
+        connect(rawView, &CanRawView::dockUndock, this, [this, rawView] { handleDock(rawView, ui->mdiArea); });
+    }
 }
 
-MainWindow::~MainWindow() {}
+void handleWidgetDeletion(QWidget* widget)
+{
+    if (widget->parentWidget()) {
+
+        widget->parentWidget()->close();
+    }
```

Comment:
I would add `assert(nullptr != widget);`. Do we know what to do in the else path? Should we report an error, discard else path silently, etc? If the else path here shall be discarded silently, please add a comment that says so.